astropy/tests/helper.py

```diff
@@ -604,3 +604,40 @@ def check_pickling_recovery(original, protocol):
     class_history = [original.__class__]
     generic_recursive_equality_test(original, unpickled,
                                     class_history)
+
+
+def assert_quantity_allclose(actual, desired, rtol=1.e-7, atol=0, err_msg='', verbose=True):
+    """
+    Raise an assertion if two objects are not equal up to desired tolerance.
```

Comment:
You should probably note in the docstring here that the definition of ``atol`` is set by the unit of `actual`, not `desired` (because `desired` gets converted to `actual.units`). Or even go a bit further and change the name of ``atol`` to ``atol_actualunit`` or something like that.